### PR TITLE
Feat(plugins): Schema support for 'convert_to_lower_case'

### LINKED ISSUE #2671: Schema validations with case_sensitivity: false is not working
Schema validations with case_sensitivity: false is not working with valid values

```
'Validation Error: underlay_routing_protocol': 'OSPF' is not one of ['ebgp', 'ospf',...]
```

_Originally posted by @carlbuchmann in https://github.com/aristanetworks/ansible-avd/pull/2520#discussion_r1152643177_

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Feat(plugins): Schema support for `convert_to_lower_case`

## Related Issue(s)

Fixes #2671 

## Component(s) name

AVD Schema tooling

## Proposed changes

- Remove the schema option `case_sensitive` which was never implemented.
- Implement the schema option `convert_to_lower_case` for string values.
  The string value will be "lowered" during conversion phase
- Update `eos_designs` schema fragments for routing protocols, where acronyms are used.

## How to test

Several of the molecule scenarios like `example-single-dc-l3ls` failed before this PR, because of upper case values.

After implementing this feature and updating the schema, the scenarios no longer fail.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/underlay_routing_protocol.schema.yml [hunk 1]
```diff
@@ -27,4 +28,4 @@ keys:
       - "isis-ldp"
       - "isis-sr-ldp"
       - "ospf-ldp"
-      - "BGP" # Only supported for historic reasons - The code will change to the default
+      - "bgp" # Only supported for historic reasons - The code will change to the default
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/underlay_routing_protocol.schema.yml]: do we still need `bgp`? if so could the commend make it to the doc?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/underlay_routing_protocol.schema.yml]: there is another pr incoming from Carl, which will remove the need for this. So everything is up in the air these days :) Please just let this one in first.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/underlay_routing_protocol.schema.yml]: Correction, let's get #2691 in first, and then we rebase this one.

**Review by gmuloc — APPROVED** [review]

**Pull request merged**